src/screens/Messages/Conversation/MessagesList.tsx

```diff
@@ -72,36 +74,63 @@ function keyExtractor(item: ConvoItem) {
   return item.key
 }
 
-function onScrollToEndFailed() {
+function onScrollToIndexFailed() {
   // Placeholder function. You have to give FlatList something or else it will error.
 }
 
 export function MessagesList() {
   const chat = useChat()
   const flatListRef = useRef<FlatList>(null)
-  // We use this to know if we should scroll after a new clop is added to the list
-  const isAtBottom = useRef(false)
-  const currentOffset = React.useRef(0)
 
-  const onContentSizeChange = useCallback(() => {
-    if (currentOffset.current <= 100) {
-      flatListRef.current?.scrollToOffset({offset: 0, animated: true})
-    }
-  }, [])
+  // We need to keep track of when the scroll offset is at the bottom of the list to know when to scroll as new items
+  // are added to the list. For example, if the user is scrolled up to 1iew older messages, we don't want to scroll to
+  // the bottom.
+  const isAtBottom = React.useRef(true)
 
-  const onEndReached = useCallback(() => {
-    if (chat.status === ConvoStatus.Ready) {
-      chat.fetchMessageHistory()
-    }
-  }, [chat])
+  // contentOpacity is a web-only hack. There is a slight delay between the rendering of the items and the user being
+  // scrolled to the bottom of the view. If we don't hide the items, there's a jarring flash whenever the scroll
+  // happens. Therefore, let's not show the items until that first scroll has actually happened
+  const [hasInitiallyScrolled, setHasInitiallyScrolled] = React.useState(false)
```

Comment:
Whenever the list was inverted, we didn't need to keep track of this state. We were guaranteed to always start at the "bottom" since...the bottom was really the top. We could also simply rely on `onEndReached` instead of `onStartReached` when adding items.

Now however, we need to keep track of this state for two reasons:

1. We do not want to call `onStartReached` on the initial render. Since the offset will be `50` (remember, `MaybeLoader` will always be rendered), the `FlatList` will fire `onStartReached` immediately. This prevents that from happening until some scroll has happened.

2. Unfortunately, we cannot rely on `initialScrollIndex` to properly place us at the right content offset - even if we let the content load before rendering the `FlatList`. `initialScrollIndex` like `scrollToIndex` is known to be unreliable unless used in combination with `getItemLayout`. In fact, the docs tell us that it cannot be used without `getItemLayout` at all https://reactnative.dev/docs/flatlist#initialscrollindex

Because of this limitation, we are going to need to rely on `scrollToOffset()` to put us at the bottom of the scroll view as soon as the content is loaded. This isn't a perfect solution, but unfortunately seems to be the one we need to rely on. Out of the box, this works fine on native. We simply use `hasInitiallyScrolled` to determine if we want to animate the scroll - we don't if it's the first scroll, then all subsequent ones should be animated.

On web however, we _additionally_ need to use this value to control whether the content is actually visible or not - using `opacity`. The content actually will appear _before_ the scroll is performed, which causes a janky flash. See below for what we do: scroll to the appropriate offset and then make the content visible.